client/src/pages/MySitters/MySitters.tsx

```diff
@@ -1,33 +1,182 @@
 /* eslint-disable prettier/prettier */
 import { useState, useEffect } from "react";
 import getRequests from '../../helpers/APICalls/getRequests';
+import updateRequest from '../../helpers/APICalls/updateRequest';
 import { RequestData } from "../../interface/Request";
-import { Container, Grid, Box } from '@material-ui/core';
+import { 
+  Container, 
+  Grid, 
+  Box, 
+  Card, 
+  CardHeader, 
+  CardContent, 
+  IconButton,
+  Typography 
+} from '@material-ui/core';
+import { Settings } from "@material-ui/icons";
 import useStyles from "./useStyles";
+import BookingCard from "../../components/BookingCards/BookingCard";
+import BookingsDatePicker from "../../components/BookingsDatePicker/BookingsDatePicker";
+
+const mockImg = 'https://badairies.co.uk/assets/admin/plugins/images/users/4.jpg';
 
 export default function MySitters(): JSX.Element {
   const classes = useStyles();
 
   const [requests, setRequests] = useState<RequestData[]>([]);
+  const [dates, setDates] = useState<Date[]>([])
+  const [cardAction, setCardAction] = useState<boolean>(false);
+  const [selectedBooking, setSelectedBooking] = useState<string>("");
 
   useEffect(() => {
     getRequests()
-      .then(data => setRequests(data));
+      .then(data => {
+        setRequests(data)
+        const dates: Date[] = [];
+        data.map(booking => {
+          if (booking.accepted) {
+            const bookingDate = new Date(booking.start)
+            dates.push(bookingDate)
+          }
+        });
+        setDates(dates);
+    });
   }, []);
+  
+  const currentDate = new Date();
+
+  const nextBookingFinder = (bookings: RequestData[]) => {
+    const nextBooking: RequestData[] = []
+    bookings.forEach((booking) => {
+      if (booking.accepted && !nextBooking.length) {
+        nextBooking.push(booking)
+      }
+    });
+    return nextBooking;
+  };
+
+  const currentBookings = requests.filter((request) => {
+    const requestDate = new Date(request.start);
+    return requestDate >= currentDate;
+  });  
+  
+  const nextBooking: RequestData[] = nextBookingFinder(currentBookings);
+
+  const pastBookings = requests.filter((request) => {
+    const requestDate = new Date(request.start);
+    return (requestDate < currentDate && request.accepted);
+  })
+
+  // function to display bookings dates
+  const createDate = (dateStart: any, dateEnd: any) => {
+    dateStart = new Date(dateStart);
+    dateEnd = new Date(dateEnd);
+    const day = dateStart.getDate();
+    let month = dateStart.toLocaleDateString("us-US", {month: "long"});
+    month = month.charAt(0).toUpperCase() + month.slice(1);
+    const year = dateStart.getFullYear();
+    const timeStart = dateStart.getHours();
+    const timeEnd = dateEnd.getHours();
+    return `${day}, ${month} ${year}, ${timeStart}-${timeEnd} HS.`
+  }
```

Comment:
Perhaps this would help clean up this code more? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat